test/index.js

```diff
@@ -4,7 +4,6 @@ const TestServer = require('./fixtures/server.js')
 const fetch = require('../lib/index.js')
 const stringToArrayBuffer = require('string-to-arraybuffer')
 const URLSearchParamsPolyfill = require('@ungap/url-search-params')
-const { URL } = require('whatwg-url')
```

Comment:
there's a test around line 1531 that uses the `URL` constructor, it used to be specific for the whatwg-url module but since the node core URL constructor is whatwg compatible the test isn't really necessary any more. that said, it's referencing `URL` as a global if we delete this line.

I think instead we should:
- [x] remove this line
- [ ] change the url require below to dereference `URL` directly instead of renaming it to `CoreURL`
- [ ] drop the test that references 'WHAWG URL' in its name
- [ ] change the test that references `Node.js URL` to use `URL` instead of `CoreURL`

at the end of that, we've completely removed whatwg-url and all references of it while not inadvertently relying on the `URL` global.

incidentally, i think the URLSearchParams polyfill can probably be removed in a follow up test too

Reply:
Yep

```js
~/D/n/minipass-fetch (gar/unused-whatwg|✚1) $ nvm use v12.13.0
Now using node v12.13.0 (npm v8.1.2)
~/D/n/minipass-fetch (gar/unused-whatwg|✚1) $ node
Welcome to Node.js v12.13.0.
Type ".help" for more information.
> url.URLSearchParams
[Function: URLSearchParams]
```

Reply:
Instead of using `url.URL` i just used `URL`

Reply:
i suppose our minimum bar is guaranteed to expose it as a global so that won't be an issue, unless a new major of node drops the global for some reason which seems unlikely